File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/server/level/ServerLevelConfiguration.java
package pize.tests.voxelgame.server.level;

import pize.math.vecmath.vector.Vec2f;
import pize.tests.voxelgame.main.level.LevelConfiguration;
import pize.tests.voxelgame.server.chunk.gen.ChunkGenerator;

public class ServerLevelConfiguration extends LevelConfiguration{
    
    private ChunkGenerator generator;
    private Vec2f spawn;
    private int seed;
    
    
    public void load(String name, int seed, ChunkGenerator generator){
        super.load(name);
        
        this.seed = seed;
        this.generator = generator;
        this.spawn = new Vec2f(-717, -952);
    }
    
    public ChunkGenerator getGenerator(){
        return generator;
    }
    
    public Vec2f getWorldSpawn(){
        return spawn;
    }
    
    public void setWorldSpawn(double x, double z){
        spawn.set(x, z);
    }
    
    public int getSeed(){
        return seed;
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/server/level/ServerLevelConfiguration.kt
package pize.tests.voxelgame.server.level

import pize.math.vecmath.vector.Vec2f
import pize.tests.voxelgame.main.level.LevelConfiguration
import pize.tests.voxelgame.server.chunk.gen.ChunkGenerator

class ServerLevelConfiguration : LevelConfiguration() {
    var generator: ChunkGenerator? = null
        private set
    var worldSpawn: Vec2f? = null
        private set
    var seed = 0
        private set

    fun load(name: String?, seed: Int, generator: ChunkGenerator?) {
        super.load(name)
        this.seed = seed
        this.generator = generator
        worldSpawn = Vec2f(-717f, -952f)
    }

    fun setWorldSpawn(x: Double, z: Double) {
        worldSpawn!![x] = z
    }
}
